frontend/vue/components/UserAccount/UserAccountLayout.vue

```diff
@@ -0,0 +1,196 @@
+<template>
+  <div class="user-account">
+    <nav class="user-account__section-nav">
+      <div class="user-account__section-nav__user-data">
+        <span class="user-account__section-nav__user-data__name">{{ userName }}</span>
+        <span class="user-account__section-nav__user-data__role">{{ userRole }}</span>
+      </div>
+      <div class="user-account__section-nav__link-list">
+        <AppLink
+          v-for="{displayName, hash} in sectionList"
+          :key="hash"
+          class="user-account__section-nav__link-list__link"
+          :class="{'user-account__section-nav__link-list__link_active': activeSection === hash}"
+          :url="hash"
+          target="_self"
+        >
+          {{ displayName }}
+        </AppLink>
+        <AppLink
+          class="user-account__section-nav__link-list__link"
+          url="#"
+          target="_self"
+        >
+          {{ $translate('Log Out') }}
+        </AppLink>
+      </div>
+    </nav>
+    <bx-dropdown
+      class="user-account__section-dropdown"
+      :value="activeSection"
+      @bx-dropdown-selected="switchPanel($event)"
+    >
+      <bx-dropdown-item
+        v-for="{displayName, hash} in sectionList"
+        :key="hash"
+        class="user-account__section-dropdown__item"
+        :value="hash"
+      >
+        {{ displayName }}
+      </bx-dropdown-item>
+    </bx-dropdown>
+    <section class="user-account__section-container">
+      <div v-if="activeSection === sectionList[0].hash">
+        PROGRESS SECTION
+      </div>
+      <div v-if="activeSection === sectionList[1].hash">
+        GROUPS SECTION
+      </div>
+      <div v-if="activeSection === sectionList[2].hash">
+        BADGES SECTION
+      </div>
+      <div v-if="activeSection === sectionList[3].hash">
+        PRIVACY SECTION
+      </div>
+    </section>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue-demi'
+import AppLink from '../common/AppLink.vue'
+
+export default defineComponent({
+  name: 'UserAccountLayout',
+  components: {
+    AppLink
+  },
+  props: {
+    userName: { type: String, required: false, default: 'Unknown' },
+    userRole: { type: String, required: false, default: 'Unknown' }
```

Comment:
I think that if we access account page, there is an account member with a name.

In the case of the role, y just used unknown because I think that we didn't think about it yet, probably we will have a generic name such as Student or similar.

Reply:
Sounds good!

So, I guess we would make the `userName` prop _required_ and the `userRole` optional but just don't render that section if there is no role being passed. Meaning, the `userRole` _default_ might be better represented by `undefined`.